### PR TITLE
Sharelink embedded into product page.

### PR DESCRIPTION
**Pull request by AsmahanM**
- [x] Added a 'copy URL' button on products page to copy URL to clipboard.
- [x] Product page information being push to supabase.
- [x]  'Contact owner' button added to bottom of product page.

### CODE REVIEW

File: pages/products/[id].js [hunk 1]
```diff
@@ -1,20 +1,51 @@
+import ShareLink from "../../components/ShareLink/ShareLink";
 import supabase from "../../utils/supabaseClient";
+import {useState, useEffect} from "react";
+import React from "react";
+import { withCoalescedInvoke } from "next/dist/lib/coalesced-function";
```
Comment (minju25kim) [inline pages/products/[id].js]: withCoalescedInvoke is used in this products/[id].js page?

Comment (alternadiva) [reply pages/products/[id].js]: good call! what does this do exactly?

Comment (AsmahanM) [reply pages/products/[id].js]: Thanks for pointing this out! It shouldn't be in there, it doesn't do anything it's just an error I forgot to delete. I've removed it now and everything still works fine.

File: pages/share-link.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import ShareLink from "../components/ShareLink/ShareLink";
```
Comment (alternadiva) [inline pages/share-link.js]: As we're using the ShareLink component directly on the product page (pages/products/[id].js), I believe we can delete this (pages/share-link.js) file.

File: components/ShareLink/ShareLink.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React, { useState } from "react";
+
+export default function ShareLink() {
```
Comment (alternadiva) [inline components/ShareLink/ShareLink.js:3]: Great work on the ShareLink component! 🔗

**Review by alternadiva — APPROVED** [review]

File: pages/products/[id].js [hunk 4]
```diff
@@ -1,20 +1,50 @@
...
-      <h1>this will be product</h1>
+      <h1>Products Page</h1>
+      <div>
+        <h3>{props.product[0].name}</h3>
+        <img src={props.product[0].image}/>
+        <p>{props.product[0].description}</p>
+        <p>{props.product[0].location}</p>
+        <p>{props.product[0].create_at}</p>
+        <p>{props.product[0].availability}</p>
+        <button>Contact {productOwner}</button>
```
Comment (sonianb) [inline pages/products/[id].js:35]: Nice addition!

**Comment by sonianb** [comment]
[Add share button](https://trello.com/c/3tYoD78p/34-add-share-button)

**Pull request merged**